testing/functional_class.py

```diff
@@ -1,19 +1,34 @@
+import os
+import urllib.request
 from abc import ABC, abstractmethod
 from utils import str_to_bool, str_to_list
 
 class FunctionalTest(ABC):
     """Class for running FATES functional tests"""
 
     def __init__(self, name:str, test_dir:str, test_exe:str, out_file:str,
-                 use_param_file:str, other_args:str):
+                 use_param_file:str, datm_file:str, datm_file_url:str, other_args:str):
         self.name = name
         self.test_dir = test_dir
         self.test_exe = test_exe
         self.out_file = out_file
         self.use_param_file = str_to_bool(use_param_file)
+        self.datm_file = None
         self.other_args = str_to_list(other_args)
         self.plot = False
-        
+
```

Comment:
I feel like adding `datm_file` in the base class is adding some technical debt since most of the classes don't need it. Why don't we only add it to the extended classes that need it.

We can have a new "base" class (extended from this one - ones that need a 'driver' file - not datm specifically).

i.e.:

`FunctionalTestNeedsDriverFile`. This class would have the methods to check if the driver file exists.

 And then the other classes can extend from that

Reply:
Great idea; will do.

Reply:
Done with c65df334.